include/shell.h

```diff
@@ -153,25 +154,9 @@ class DOS_Shell : public Program {
 	bool call = false;
 };
 
-/* Object to manage lines in the autoexec.bat The lines get removed from
- * the file if the object gets destroyed. The environment is updated
- * as well if the line set a a variable */
-class AutoexecObject{
-private:
-	bool installed = false;
-	std::string buf = {};
-
-public:
-	AutoexecObject() = default;
-	AutoexecObject(const std::string& line);
-	~AutoexecObject();
-
-	void Install(const std::string& in);
-	void InstallBefore(const std::string& in);
-
-private:
-	void CreateAutoexec();
-};
+// XXX
+std::tuple<std::string, std::string, std::string> parse_drive_conf(
+        std::string drive_letter, const std_fs::path& conf_path);
```

Comment:
My general view on tuples: use them very sparingly, preferably only internally in functions. To be honest, I don't like seeing them in public APIs at all... Especially not when there's more than two tuple elements.

Using new structs instead is very easy and that way the return types become self-documenting. Tuples are just returning a bunch of values in a bucket; then you need a lot of explaining to do via comments, which get ouf of sync easily...

Tuples can be cool in purely functional languages, but the value is questionable in C++, Kotlin, Scala, etc. I'd 100% use a struct in this case that has meaningful names for these three values. In C++, I see them as a quick'n'dirty shortcut—again, that's why I'd only use it in some internal lambdas, personally.

Reply:
That's not me - this is a legacy code, not directly having much to do with my rework. It's marked as changed only because the clang format script decided to alter it a little.

Reply:
I guess you could optionally still improve it, but ok :)